libraries/AP_ESC_Telem/AP_ESC_Telem.cpp

```diff
@@ -156,6 +145,22 @@ bool AP_ESC_Telem::get_motor_temperature(uint8_t esc_index, int16_t& temp) const
     return true;
 }
 
+// get the highest ESC temperature in centi-degrees if available, returns true if there is valid data for at least one ESC
+bool AP_ESC_Telem::get_highest_motor_temperature(int16_t& temp) const
+{
+    uint8_t valid_escs = 0;
+
+    for (uint8_t i = 0; i < ESC_TELEM_MAX_ESCS; i++) {
+        int16_t temp_temp;
+        if (get_motor_temperature(i, temp_temp)) {
+            temp = MAX(temp, temp_temp);
+            valid_escs++;
+        }
+    }
+
+    return !!valid_escs; // return true if valid_escs > 0
```

Comment:
```suggestion
    return valid_escs > 0;
```

If it needs a comment.... :-)

Reply:
So first you ask me to add a comment then you ask me to change it because it needed a comment. You guys... :rofl: